dig.go

```diff
@@ -372,10 +395,28 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 			return err
 		}
 	}
+	c.addNode(n)
 
 	return nil
 }
 
+// addNode adds the dotEdges in node n into the DOT-format graph in Container c.
+func (c *Container) addNode(n *node) {
```

Comment:
nit: Seeing as node is frequently referenced, we should be a _little_ more explicit with what this method is doing. We could make a method on the `dotGraph` type so that it's more clearly insulated.

```
func (dg *dotGraph) add(n *node) { ... }
```

The above call would then become `c.dg.add(n)`